### PR TITLE
Svc typenames star traffic policy type

### PR DESCRIPTION
**Pull request by thockin**
Just a bit of redundancy removal.  Old names left for compat.

/kind cleanup

```release-note
NONE
```
 /sig network

### CODE REVIEW

**Comment by thockin** [comment]
Rebased

**Comment by aojea** [comment]
/retest
CSI tests failures with the following errors

```
{ Failure test/e2e/framework/framework.go:187
Jul  7 16:59:50.853: All nodes should be ready after test, Get "https://127.0.0.1:45621/api/v1/nodes": dial tcp 127.0.0.1:45621: connect: connection refused
vendor/github.com/onsi/ginkgo/internal/leafnodes/runner.go:113}
```

File: pkg/generated/openapi/zz_generated.openapi.go [hunk 1]
```diff
@@ -24930,10 +24930,10 @@ func schema_k8sio_api_core_v1_ServiceSpec(ref common.ReferenceCallback) common.O
...
 							Type:        []string{"string"},
 							Format:      "",
-							Enum:        []interface{}{"Cluster", "Local"}},
+							Enum:        []interface{}{"Cluster", "Cluster", "Local", "Local"}},
```
Comment (aojea) [inline pkg/generated/openapi/zz_generated.openapi.go:24936]: is this ok?

**Comment by aojea** [comment]
> @thockin: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> 
> Test name	Commit	Details	Required	Rerun command
> pull-kubernetes-e2e-kind	[74d058e]

ref https://github.com/kubernetes/kubernetes/issues/111086

**Comment by danwinship** [comment]
/approve
but needs rebase

**Comment by thockin** [comment]
Rebased and updated

File: staging/src/k8s.io/api/core/v1/types.go [hunk 2]
```diff
@@ -4270,20 +4270,28 @@ const (
 	ServiceInternalTrafficPolicyLocal ServiceInternalTrafficPolicyType = "Local"
 )
 
-// ServiceExternalTrafficPolicyType describes how nodes distribute service traffic they
+// ServiceExternalTrafficPolicy describes how nodes distribute service traffic they
 // receive on one of the Service's "externally-facing" addresses (NodePorts, ExternalIPs,
-// and LoadBalancer IPs).
```
Comment (aojea) [inline staging/src/k8s.io/api/core/v1/types.go:4275]: is this on purpose? or you delete this line accidentally?

Comment (thockin) [reply staging/src/k8s.io/api/core/v1/types.go:4275]: nope, mistake

**Comment by thockin** [comment]
comment fixed and repushed

**Comment by thockin** [comment]
ping @aojea

**Comment by aojea** [comment]
/assign @andrewsykim 

for the missing approval

**Review by andrewsykim — COMMENTED** [review]
/approve

for legacy-cloud-providers changes

**Comment by aojea** [comment]
we can wait for 1.27?

**Comment by thockin** [comment]
yeah, it's not urgent

**Comment by aojea** [comment]
> It is, however, tedious to rebase :)

_slacker_ XDDD

**Comment by thockin** [comment]
Rebased

**Comment by dims** [comment]
If you still need this PR then please rebase, if not, please close the PR

**Comment by ameukam** [comment]
/milestone v.127

The milestone_applier plugin was not configured to add 1.27 milestone

**Pull request merged**